packages/core/src/tracing/cls.js

```diff
@@ -549,31 +552,50 @@ function skipExitTracing(options) {
       isActive: true,
       extendedResponse: false,
       skipParentSpanCheck: false,
-      skipIsTracing: false
+      skipIsTracing: false,
+      checkReducedSpan: false
     },
     options
   );
 
-  const parentSpan = getCurrentSpan();
+  let parentSpan = getCurrentSpan();
+
+  // If there is no active entry span, we fall back to the reduced span of the most recent entry span.
+  // See comment in packages/core/src/tracing/clsHooked/unset.js#storeReducedSpan.
+  if (opts.checkReducedSpan) {
+    parentSpan = getCurrentSpan() || getReducedSpan();
+  }
+
   const suppressed = tracingSuppressed();
   const isExitSpanResult = isExitSpan(parentSpan);
 
   // CASE: first ask for suppressed, because if we skip the entry span, we won't have a parentSpan
   //       on the exit span, which would create noisy log messages.
   if (suppressed) {
-    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult };
+    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult, parentSpan };
     return true;
   }
 
+  if (allowRootExitSpan) {
+    if (opts.extendedResponse) return { skip: false, suppressed, isExitSpan: isExitSpanResult, parentSpan };
+    else return false;
+  }
+
+  if (opts.checkReducedSpan && (!parentSpan || isExitSpan(parentSpan))) {
```

Comment:
suggestion: These conditions become quite complex. Could you please describe the cases + why? Then it will be easier to review 👍